src/Fixer/PhpTag/ShortEchoTagFixer.php

```diff
@@ -0,0 +1,242 @@
+                ->setDefault(false)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::FORMAT_SHORT === $this->configuration[self::OPTION_FORMAT]) {
+            $this->longToShort($file, $tokens);
+        } else {
+            $this->shortToLong($file, $tokens);
+        }
+    }
+
+    private function longToShort(SplFileInfo $file, Tokens $tokens)
+    {
+        $offset = 0;
+        for (;;) {
+            $foundEcho = $tokens->findSequence([[\T_OPEN_TAG], [\T_ECHO]], $offset);
+            $foundPrint = $tokens->findSequence([[\T_OPEN_TAG], [\T_PRINT]], $offset);
+            if (null === $foundEcho) {
+                $found = $foundPrint;
+                if (null === $found) {
+                    break;
+                }
+            } elseif (null === $foundPrint || key($foundEcho) < key($foundPrint)) {
+                $found = $foundEcho;
+            } else {
+                $found = $foundPrint;
+            }
+            $startRange = key($found);
+            $offset = $startRange + 1;
+            if (!$this->configuration[self::OPTION_SHORT_ALWAYS] && $this->isComplexCode($tokens, $offset)) {
+                continue;
+            }
+            next($found);
+            $endRange = key($found);
+            $newTokens = $this->buildLongToShortTokens($tokens, $startRange, $endRange);
+            $tokens->overrideRange($startRange, $endRange, $newTokens);
+        }
+    }
+
+    private function shortToLong(SplFileInfo $file, Tokens $tokens)
+    {
+        if (self::LONG_FUNCTION_PRINT === $this->configuration[self::OPTION_LONG_FUNCTION]) {
+            $echoToken = [\T_PRINT, 'print'];
+        } else {
+            $echoToken = [\T_ECHO, 'echo'];
+        }
+        $offset = $tokens->count() - 1;
+        for (;;) {
+            $found = $tokens->getPrevTokenOfKind($offset, [[\T_OPEN_TAG_WITH_ECHO]]);
+            if (null === $found) {
+                break;
+            }
+            $offset = $found - 1;
+            $replace = [new Token([\T_OPEN_TAG, '<?php ']), new Token($echoToken)];
+            if (!$tokens[$found + 1]->isWhitespace()) {
+                $replace[] = new Token([\T_WHITESPACE, ' ']);
+            }
+            $tokens->overrideRange($found, $found, $replace);
+        }
+    }
+
+    private function isComplexCode(Tokens $tokens, $index)
+    {
+        $semicolonFound = false;
+        for ($count = $tokens->count(); $index < $count; ++$index) {
+            $token = $tokens[$index];
+            if ($token->isGivenKind(\T_CLOSE_TAG)) {
+                return false;
+            }
+            if (';' === $token->getContent()) {
+                $semicolonFound = true;
+            } elseif ($semicolonFound && !$token->isWhitespace()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
```

Comment:
No need for FQCN in this PHPDoc (BTW seems like an idea for new fixer).

Reply:
I do prefer FQCN in PHPDocs, and that's for these reasons:

1. If we need to reference a class only in PHPDocs, do we need to add a `use` statement at the beginning of the file?
If yes, it's quite useless to me.
If no: what if we add at a later time a reference to it in the code: should we fix all the previous PHPDocs?

2. consider this example:
    ```
    <?php
    use Namespace1\ClassName;
    use Namespace2\ClassName as ClassNameAlias;
    
    /**
     * @return ClassNameAlias
     */
    function foo() {
    }
    ```
    In order to know what `ClassNameAlias` actually is, humans have to read the `use` statements at the top of the file.

BTW, since this project doesn't use FQCN in comments, I updated this class: 9031768b641b9fb3ae82ea0b48053d0fd28cd5fe